packages/typegpu/src/data/struct.ts

```diff
@@ -20,7 +21,14 @@ import type { AnyWgslData, WgslStruct } from './wgslTypes.ts';
 export function struct<TProps extends Record<string, AnyWgslData>>(
   props: TProps,
 ): WgslStruct<TProps> {
-  const struct = <T>(props: T) => props;
+  // in the schema call, create and return a deep copy
+  // by wrapping all the values in corresponding schema calls
+  const struct = <T extends {}>(instanceProps: T) =>
+    Object.fromEntries(
+      Object.entries(instanceProps).map((
+        [key, value],
+      ) => [key, schemaCallWrapper(props[key], value)]),
+    );
```

Comment:
Let's iterate over the `props` instead of `instanceProps`, to automatically remove any unwanted keys without causing an issue.